interface/reports/patient_list_creation.php

```diff
@@ -246,582 +319,773 @@ function srch_option_change(elem) {
+        $whr_stmt .= " and pd.sex = ?";
+        array_push($sqlBindArray, $sql_gender);
+    }
+    if (strlen($sql_ethnicity) != 0) {
+        $whr_stmt .= " and pd.ethnicity = ?";
+        array_push($sqlBindArray, $sql_ethnicity);
+    }
 
-            if (strlen($sql_gender) != 0) {
-                $whr_stmt = $whr_stmt . "   and pd.sex = ?";
-                array_push($sqlBindArray, $sql_gender);
-            }
+    // WHERE conditions based on inputs arising from specific search options
+    if ($srch_option == "Prescriptions" && strlen($prescription_drug) > 0) {
+        $whr_stmt .= " AND rx.drug LIKE ?";
+        array_push($sqlBindArray, $prescription_drug);
+    }
+    if ($srch_option == "Communication" && strlen($communication) > 0) {
+        if ($communication == "allow_sms") {
+            $whr_stmt .= " AND pd.hipaa_allowsms = 'YES' ";
+        } elseif ($communication == "allow_voice") {
+            $whr_stmt .= " AND pd.hipaa_voice = 'YES' ";
+        } elseif ($communication == "allow_mail") {
+            $whr_stmt .= " AND pd.hipaa_mail  = 'YES' ";
+        } elseif ($communication == "allow_email") {
+            $whr_stmt .= " AND pd.hipaa_allowemail  = 'YES' ";
+        }
+    }
+    if ($srch_option == "Insurance Companies" && strlen($insurance_company) > 0 && $insurance_company != "All") {
+        $whr_stmt .= " AND id.provider = ?";
+        array_push($sqlBindArray, $insurance_company);
+    }
+    if ($srch_option == "Encounters" && strlen($encounter_type) > 0 && $encounter_type != "All") {
+        $whr_stmt .= " AND enc.encounter_type_code = ?";
+        array_push($sqlBindArray, $encounter_type);
+    }
+    if ($srch_option == "Observations" && strlen($observation_description) > 0) {
+        $whr_stmt .= " AND (obs.code LIKE ? OR obs.description LIKE ?)";
+        array_push($sqlBindArray, $observation_description, $observation_description);
+    }
+    if (strlen($procedure_diagnosis) > 0) {
+        if ($srch_option == "Allergies" || $srch_option == "Problems" || $srch_option == "Medications") {
+            $whr_stmt .= " AND li.diagnosis LIKE ?";
+            array_push($sqlBindArray, $procedure_diagnosis);
+        } else if ($srch_option == "Procedures" || $srch_option == "Lab Results") {
+            $whr_stmt .= " AND (pr_ord.order_diagnosis LIKE ? OR pr_code.diagnoses LIKE ?)";
+            array_push($sqlBindArray, $procedure_diagnosis, $procedure_diagnosis);
+        }
+    }
 
-            if (strlen($sql_ethnicity) != 0) {
-                $whr_stmt = $whr_stmt . "   and pd.ethnicity = ?";
-                array_push($sqlBindArray, $sql_ethnicity);
-            }
+    // Controls the columns displayed, their headings and widths, and how many columns are sorted from the left
+    $report_options_arr = array(
+        "Demographics" => array(
+            "cols" => array(
+                "patient_date"   => array("heading" => "Date Created", "width" => "nowrap"),
```

Comment:
if heading is only used as a label, then they should all be translated via xl here (so these constants will then get picked up via OpenEMR's translation pipeline)
And then below where you output the heading, would just escape it via text() (or attr() if in a html attribute)